services/services.go

```diff
@@ -53,6 +65,9 @@ func NewServiceReplication() ServiceReplication { return new(serviceReplication)
 type ServiceSharding interface {
 	// Len is the count of shards to use as the sharding range
 	Len() int
```

Comment:
``Len`` seems like a misleading terminology for this.  In go-land ``Len`` typically denotes the number of items in a container, in this case its really the cardinality of a numeric space which seems more like a ``Count`` or (to use Google style) ``NumShards``.